lib/model.js

```diff
@@ -1009,6 +1026,39 @@ Connection = AmpersandModel.extend({
     if (!attrs.ssh_tunnel_port) {
       throw new TypeError('ssl_tunnel_port is required when ssh_tunnel is not NONE.');
     }
+  },
+  validate_stitch: function(attrs) {
+    if (attrs.connectionType === CONNECTION_TYPE_VALUES.STITCH_ATLAS) {
+      if (!attrs.stitchClientAppId) {
+        throw new TypeError('stitchClientAppId is required when connectionType is STITCH_ATLAS.');
+      }
+    } else if (attrs.connectionType === CONNECTION_TYPE_VALUES.STITCH_ON_PREM) {
+      if (!attrs.stitchClientAppId) {
+        throw new TypeError('stitchClientAppId is required when connectionType is stitchOnPrem.');
```

Comment:
`stitchOnPrem` => `STITCH_ON_PREM` 

Optional - consider using the ES6 template strings instead, i.e. `${CONNECTION_TYPE_VALUES.STITCH_ON_PREM`

Also repeat for the three other messages below.